DOCSP-28808: conversion operations (#358)
mongodb/docs-java: source/fundamentals/aggregation/aggregation-expression-operations.txt

                                      else: "no advisory" } } } }

Conversion Operations
~~~~~~~~~~~~~~~~~~~~~

You can perform a conversion operation to convert between ``MqlValue``
types using the methods described in this section.

.. list-table::
   :header-rows: 1
   :widths: 40 60

   * - Java Method
     - Aggregation Pipeline Operator

   * - `asDocument() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#asDocument()>`__
     - :manual:`$convert </reference/operator/aggregation/convert/>`
       
   * - `asMap() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#asMap()>`__
     - :manual:`$convert </reference/operator/aggregation/convert/>`

   * - `asString() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#asString()>`__
     - :manual:`$toString </reference/operator/aggregation/toString/>`

   * - `asString() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#asString(com.mongodb.client.model.mql.MqlString,com.mongodb.client.model.mql.MqlString)>`__
     - :manual:`$dateToString </reference/operator/aggregation/dateToString/>`

>    * - `parseDate() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#parseDate()>`__
>      - :manual:`$dateFromString </reference/operator/aggregation/dateFromString/>`
> 
>    * - `parseInteger() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#parseInteger()>`__
>      - :manual:`$toInt </reference/operator/aggregation/toInt/>`
> 
>    * - `millisecondsAsDate() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#millisecondsAsDate()>`__
>      - :manual:`$toDate </reference/operator/aggregation/toDate/>`

Reviewer: I'd recommend putting these in alphabetical order so it is easier for readers to follow.

```suggestion
   * - `millisecondsAsDate() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#millisecondsAsDate()>`__
     - :manual:`$toDate </reference/operator/aggregation/toDate/>`

   * - `parseDate() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#parseDate()>`__
     - :manual:`$dateFromString </reference/operator/aggregation/dateFromString/>`

   * - `parseInteger() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#parseInteger()>`__
     - :manual:`$toInt </reference/operator/aggregation/toInt/>`
```

.. include:: /includes/fundamentals/agg-mql-diff.rst

> Suppose you want to calculate students' fifth reunion years based on
> their graduation years, which are stored as strings. You want to add a

Reviewer: [suggestion]

For readability, I'd recommend a slight rewording. You could also break this into two sentences, but that's totally up to you.

```suggestion
Suppose you have a collection of students with their graduation years
stored as strings and want to calculate the years for their fifth reunions.
You want to add a
```
